packages/amplify-cli/src/extensions/amplify-helpers/resource-status-data.ts

```diff
@@ -0,0 +1,536 @@
+    allResources = allResources.filter(resource => resource.providerPlugin === providerName);
+  }
+  // if not equal there is a tag update
+  const tagsUpdated = !_.isEqual(stateManager.getProjectTags(), stateManager.getCurrentProjectTags());
+
+  return {
+    resourcesToBeCreated,
+    resourcesToBeUpdated,
+    resourcesToBeSynced,
+    resourcesToBeDeleted,
+    tagsUpdated,
+    allResources,
+  };
+}
+
+export function getAllResources(amplifyMeta, category, resourceName, filteredResources) {
+  let resources: any[] = [];
+
+  Object.keys(amplifyMeta).forEach(categoryName => {
+    const categoryItem = amplifyMeta[categoryName];
+    Object.keys(categoryItem).forEach(resource => {
+      amplifyMeta[categoryName][resource].resourceName = resource;
+      amplifyMeta[categoryName][resource].category = categoryName;
+      resources.push(amplifyMeta[categoryName][resource]);
+    });
+  });
+
+  resources = filterResources(resources, filteredResources);
+
+  if (category !== undefined && resourceName !== undefined) {
+    // Create only specified resource in the cloud
+    resources = resources.filter(resource => resource.category === category && resource.resourceName === resourceName);
+  }
+
+  if (category !== undefined && !resourceName) {
+    // Create all the resources for the specified category in the cloud
+    resources = resources.filter(resource => resource.category === category);
+  }
+
+  return resources;
+}
+
+export function getResourcesToBeCreated(amplifyMeta, currentAmplifyMeta, category, resourceName, filteredResources) {
+  let resources: any[] = [];
+
+  Object.keys(amplifyMeta).forEach(categoryName => {
+    const categoryItem = amplifyMeta[categoryName];
+    Object.keys(categoryItem).forEach(resource => {
+      if (
+        (!amplifyMeta[categoryName][resource].lastPushTimeStamp ||
+          !currentAmplifyMeta[categoryName] ||
+          !currentAmplifyMeta[categoryName][resource]) &&
+        categoryName !== 'providers' &&
+        amplifyMeta[categoryName][resource].serviceType !== 'imported'
+      ) {
+        amplifyMeta[categoryName][resource].resourceName = resource;
+        amplifyMeta[categoryName][resource].category = categoryName;
+        resources.push(amplifyMeta[categoryName][resource]);
+      }
+    });
+  });
+
+  resources = filterResources(resources, filteredResources);
+
+  if (category !== undefined && resourceName !== undefined) {
+    // Create only specified resource in the cloud
+    resources = resources.filter(resource => resource.category === category && resource.resourceName === resourceName);
+  }
+
+  if (category !== undefined && !resourceName) {
+    // Create all the resources for the specified category in the cloud
+    resources = resources.filter(resource => resource.category === category);
+  }
+
+  // Check for dependencies and add them
+
+  for (let i = 0; i < resources.length; ++i) {
```

Comment:
Can we changed it to ForEach() , just like in other functions if time permits?

Reply:
This seems to be a dual iteration for dependencies. will refactor this separately.

Reply:
This code was essentially one part of a 3 part split of the resource-status.ts file (legacy summary table), hence it didn't have some of the best practices which implemented in the earlier files. Have refactored it to use constants, common functions, for each among other things.